plugins/gossip/plugin.go

```diff
@@ -138,11 +143,40 @@ func configureMessageLayer() {
 		}
 
 		msg := cachedMsgEvent.Message.Unwrap()
+
+		// do not gossip requested messages
+		requestedMsgMutex.Lock()
+		if _, exist := requestedMsgs[msg.ID()]; exist {
+			delete(requestedMsgs, msg.ID())
```

Comment:
Why do we delete here? Isn't going to then forget we've requested that msg?

Reply:
Yes, it is. Since the msg is solid and already stored, the msg shouldn't trigger `MessageSolid` event and be gossiped anymore, no?

Reply:
You are right!